packages/integrations/src/transactionactivity.ts

```diff
@@ -0,0 +1,152 @@
+import { EventProcessor, Hub, Integration, Scope, Span, SpanContext } from '@sentry/types';
+
+/** JSDoc */
+interface TransactionActivityOptions {
+  // onLocationChange: (info) => {
+  //   // info holds the location change api.
+  //   // if this returns `null` there is no transaction started.
+  //   return info.state.transaction || info.url;
+  // },
+  // onActivity?: (info) => {
+  //   return info.type !== 'xhr' || !info.url.match(/zendesk/);
+  // },
+  idleTimeout?: number;
+}
+
+/** JSDoc */
+interface Activity {
+  name: string;
+  span?: Span;
+}
+
+/** JSDoc */
+export class TransactionActivity implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public name: string = TransactionActivity.id;
+
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'TransactionActivity';
+
+  /** JSDoc */
+  private static _options: TransactionActivityOptions;
+
+  /**
+   * Returns current hub.
+   */
+  private static _getCurrentHub?: () => Hub;
+
+  private static _activeTransaction?: Span;
+
+  private static _currentIndex: number = 0;
+
+  private static readonly _activities: { [key: number]: Activity } = {};
+
+  private static _debounce: number = 0;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options?: TransactionActivityOptions) {
+    TransactionActivity._options = {
+      idleTimeout: 500,
+      ...options,
+    };
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    TransactionActivity._getCurrentHub = getCurrentHub;
+  }
+
+  /**
+   * Internal run loop that checks if activy is running
+   */
+  private static _watchActivity(): void {
+    const count = Object.keys(TransactionActivity._activities).length;
```

Comment:
I wonder if we shouldnt use an array here. Object prevents us from having multiple activities with the same name. And we use `pop` with the reference anyway, so array would be fine as well.

Reply:
Although I see you use index as the key here already. You can ignore this comment.

Reply:
Yes, also JS arrays.
```
let arr = [1,2,3];
> undefined
arr.length
> 3
delete arr[1]
> true
arr.length
> 3
arr
> (3) [1, empty, 3]

 (╯°□°）╯︵ ┻━┻) 
```